impeller/tools/impeller.gni

```diff
@@ -349,17 +349,19 @@ template("impellerc") {
 
     shader_target_flag = invoker.shader_target_flag
 
-    depfile_path = "$generated_dir/{{source_file_part}}.d"
-    depfile_intermediate_path = rebase_path(depfile_path, root_build_dir)
-    depfile = depfile_path
-
     shader_lib_dir = rebase_path("//flutter/impeller/compiler/shader_lib")
     args = [
       "--include=$shader_lib_dir",
-      "--depfile=$depfile_intermediate_path",
       "$shader_target_flag",
     ]
 
+    if (!shader_bundle) {
+      depfile_path = "$generated_dir/{{source_file_part}}.d"
```

Comment:
Huh, it's weird that using `source_file_part` doesn't error out here for `action`/`compiled_action` like the other source-based vars do when using single-invocation mode.

I think moving this behind the `!single_invocation` guard below would make the most sense. Even though `shader_bundle` mode is technically the only situation where we'll use `single_invocation`, we'll need to switch `--iplr` to behave similarly in the future once we change it to pack multiple backends into a single iplr at once for RuntimeEffect support on Android.

Reply:
Done

(`source_file_part` was being replaced with an empty string in the depfile path when single invocation mode invoked an `action` without a `foreach`)